### DISCUSSION TITLE
Add framework-specific documentation

### DISCUSSION THREAD

**Discussion by ashleyryan** [vmware/clarity#5895] [Ideas]
Some frameworks (aka React) require framework-specific logic to make a component work. It might be helpful to have a way in the docs of denoting framework-specific logic.

Two examples:
- You cannot set `indeterminate` directly on the `<input>` for `<CdsCheckbox>`. You have to set it in a `useLayoutEffect` using a ref on the input. 
- Modals - you don't need to add/remove attributes with event listeners, you can do it directly with react and use a `useState` to track the state of the component. 

I'm happy to contribute if there's a pattern for doing so.

**Comment by dturcotte** [comment]
Another quirk is with boolean attributes:

Example with `cds-control`: https://stackblitz.com/edit/react-ts-6qsknt?file=Example.tsx
More detail: https://github.com/facebook/react/issues/9230

**Comment by bbogdanov** [comment]
Hi @ashleyryan, 

We would appreciate some contributions for the React complications. 

I believe https://github.com/vmware/clarity/blob/next/packages/react/README.md will be a good place to start. Maybe adding a new section. 

Another way could be to create a new markdown file and the team will find a place for it on the website.
